### PR TITLE
[FEATURE] Add pull request title validation GitHub action

### PR DESCRIPTION
**Pull request by axbarsan**
Makes pull requests name require a prefix, in order to be able to automatically generate the changelog.

Prefixes can be:
* **[FEATURE]** - For a new feature
* **[CHANGE]** - For changes in existing functionality
* **[DEPRECATE]** - For soon-to-be removed features
* **[REMOVE]** - For removing features
* **[BUGFIX]** - For bug fixes
* **[SECURITY]** - For vulnerability fixes

Very much a work in progress, might need to make a couple of PRs to test it

### CODE REVIEW

**Comment by stone-z** [comment]
Hahaha it got you already! I love it

**Comment by ubergesundheit** [comment]
To which category do refactorings belong to?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: CI
+
+on: pull_request
+
+jobs:
+  update_pr:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: deepakputhraya/action-pr-title@master
```
Comment (stone-z) [inline .github/workflows/main.yml]: I wonder if there is a way we can somehow control these actions without forking and managing them ourselves. Maybe pin the specific version to start with

**Comment by axbarsan** [comment]
> To which category do refactorings belong to?

I updated the **[CHANGE]** prefix to say `for changes in EXISTING functionality`, it didn't really make sense before.

So the most sensible one would be **[CHANGE]**

Thank you!

File: .github/workflows/main.yml (hunk 1 above)
Comment (axbarsan) [inline .github/workflows/main.yml]:
```suggestion
    - uses: deepakputhraya/action-pr-title@v1
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+name: CI
+
+on: pull_request
+
+jobs:
+  update_pr:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: deepakputhraya/action-pr-title@v1
```
Comment (axbarsan) [inline .github/workflows/main.yml]:
```suggestion
    - uses: deepakputhraya/action-pr-title@v1.0.0
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+name: CI
+
+on: pull_request
+
+jobs:
+  update_pr:
```
Comment (axbarsan) [inline .github/workflows/main.yml]:
```suggestion
  validate_pr_name:
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+on: pull_request
+
+jobs:
+  validate_pr_name:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: deepakputhraya/action-pr-title@v1.0.0
+      with:
+        regex: '(\[[A-Z]\])+([a-zA-Z])+'
```
Comment (axbarsan) [inline .github/workflows/main.yml]:
```suggestion
        regex: '(\[[A-Z]\])+\s([a-zA-Z])+'
```
```suggestion
        regex: '(\[[A-Z]\])+([a-zA-Z])+'
```

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+on: pull_request
+
+jobs:
+  validate_pr_name:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: deepakputhraya/action-pr-title@v1.0.0
+      with:
+        regex: '(\[[A-Z]\])+\s([a-zA-Z])+'
```
Comment (axbarsan) [inline .github/workflows/main.yml]:
```suggestion
        regex: '(\[[A-Z]\])+\s([ a-zA-Z])+'
```

**Comment by axbarsan** [comment]
RIP my regex skills

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+on: pull_request
+
+jobs:
+  validate_pr_name:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: deepakputhraya/action-pr-title@v1.0.0
+      with:
+        regex: '(\[[A-Z]\])+\s([ a-zA-Z])+'
```
Comment (axbarsan) [inline .github/workflows/main.yml]:
```suggestion
        regex: '\[([A-Z])+\]([ a-zA-Z0-9])+'
```

**Comment by axbarsan** [comment]
VICTORY!

**Pull request merged**